db/models.py

```diff
@@ -40,8 +48,95 @@ def __str__(self) -> str:
 
 class MovieSession(models.Model):
     show_time = models.DateTimeField()
-    cinema_hall = models.ForeignKey(to=CinemaHall, on_delete=models.CASCADE)
-    movie = models.ForeignKey(to=Movie, on_delete=models.CASCADE)
+    cinema_hall = models.ForeignKey(
+        to="CinemaHall", on_delete=models.CASCADE, related_name="cinema_halls"
+    )
+    movie = models.ForeignKey(to="Movie", on_delete=models.CASCADE)
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(to="User", on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return f"{self.created_at}"
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        to="MovieSession",
+        on_delete=models.CASCADE,
+        related_name="movie_sessions",
+    )
+    order = models.ForeignKey(
+        to="Order", on_delete=models.CASCADE, related_name="orders"
+    )
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=[
+                    "row",
+                    "seat",
+                    "movie_session",
+                ],
+                name="unique_row_seat_session",
+            )
+        ]
+
+    def __str__(self) -> str:
+        return (f"{self.movie_session.movie.title} "
+                f"{self.movie_session.show_time} "
+                f"(row: {self.row}, seat: {self.seat})")
+
+    def save(
+        self,
+        force_insert: bool = False,
+        force_update: bool = False,
+        using: bool = None,
+        update_fields: bool = None,
+    ) -> Ticket:
```

Comment:
This method's type annotation is incorrect. You can refer to the signature of `super().save()` and copy it.